accepted/0000-sle-micro-reboot-needed.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: SLE Micro: report when a reboot is needed
+- Start Date: 2021-09-23
+
+# Summary
+[summary]: #summary
+
+After a system level change in SLE Micro, SUSE Manager should indicate in the UI when a reboot is needed.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, for SLE Micro, we display the following message on the detail page: `OS with transactional updates: please reboot after any packages and/or channel changes`.
+
+This is not ideal when it comes to usability. The user is responsible for knowing when a reboot is needed, and scheduling those reboots.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The SLE Micro uses transactional updates. Transactional updates are atomic (all updates are applied only if all updates succeed) and support rollbacks. They don't affect a running system as no changes are activated until after the system is rebooted.
+To be able to have up-to-date information regarding reboot needed, we need to update this information after every state apply.
+For that, this proposed solution is based on the following steps:
+ - Create an execution module to check if a reboot is required
+ - After every state apply run this execution module
+ - Store the result in a postgresql column
+ - Report reboot needed in the UI based on this column
+
+[This draft Pull Request](https://github.com/SUSE/spacewalk/pull/19036) can help to understand the general idea, and the details are explained below.
+
+### The execution module
+
+To check if a reboot is needed we can use `snapper` to determine if the default snapshot is not active. If so, a reboot is pending. One can find more details about this strategy in [Jira](https://jira.suse.com/browse/CSD-95) discussion. The execution module based on this strategy can be seen [here](https://github.com/SUSE/spacewalk/blob/3c2031bac04fbe88cd5b6692a3e511a457cdf5e7/susemanager-utils/susemanager-sls/src/modules/rebootutil.py).
```

Comment:
I think that `pending_transaction` is the correct call here ... not sure why it is not mentioned. `t_u` and `r_m` modules are complex and cover all the use cases.

Reply:
@aplanas I think you are right, it was not mentioned because I was not aware of this possibility. I will update the RFC to use `pending_transaction` to check if a reboot is needed, seems to be the best way to do it.